packages/vite-plugin/src/main.test.ts

```diff
@@ -705,6 +705,67 @@ defined on your team and site and much more. Run npx netlify init to get started
         await server.close()
         await fixture.destroy()
       })
+
+      test('Ignores SPA redirect in dev mode', async () => {
+        const fixture = new Fixture()
+          .withFile(
+            'netlify.toml',
+            `[[redirects]]
+              from = "/*"
+              to = "/index.html"
+              status = 200`,
+          )
+          .withFile(
+            'vite.config.js',
+            `import { defineConfig } from 'vite';
+             import netlify from '@netlify/vite-plugin';
+
+             export default defineConfig({
+               plugins: [
+                 netlify({
+                   middleware: true,
+                 })
+               ]
+             });`,
+          )
+          .withFile(
+            'index.html',
+            `<!DOCTYPE html>
+             <html>
+               <head><title>SPA App</title></head>
+               <body>
+                 <div id="app"></div>
+                 <script type="module" src="/src/main.js"></script>
+               </body>
+             </html>`,
+          )
+          .withFile('src/main.js', `document.getElementById('app').textContent = 'Hello from SPA'`)
+        const directory = await fixture.create()
+        await fixture
+          .withPackages({
+            vite: viteVersion,
+            '@netlify/vite-plugin': pathToFileURL(path.resolve(directory, PLUGIN_PATH)).toString(),
+          })
+          .create()
+
+        const { server, url } = await startTestServer({
+          root: directory,
+        })
+
+        // Any route should render the root index.html (Vite handles it) and JS should execute (which
+        // verifies the SPA redirect isn't interfering with loading the .js module).
+        await page.goto(`${url}/some-route`)
+        await page.waitForSelector('#app')
+        expect(await page.textContent('#app')).toBe('Hello from SPA')
+
+        // Client-side navigation should also work (Vite handles it)
+        await page.goto(`${url}/some-other-route`)
+        await page.waitForSelector('#app')
+        expect(await page.textContent('#app')).toBe('Hello from SPA')
```

Comment:
Neither `/some-route` or `/some-outher-route` is defined in fixture, so are those 2 checks basically testing same thing (that vite will serve index.html fallback and our redirect is not interfering in getting js bundle) or I'm missing something?

Can one of these be removed without losing any coverage?